metrics/collectors/economic/economic.go

```diff
@@ -135,20 +129,29 @@ func (ec *Collector) Collect(ch chan<- prometheus.Metric) {
 	// TODO(samlaf): implement this. probably have to call the BLSOperatorStateRetriever contract?
 	// probably should start using the avsregistry service instead of chainio clients so that we can
 	// swap out backend for a subgraph eventually
-	quorumNums, blsOperatorStateRetrieverOperator, err := ec.avsRegistryReader.GetOperatorsStakeInQuorumsOfOperatorAtCurrentBlock(context.Background(), ec.operatorId)
+	operatorId, err := ec.avsRegistryReader.GetOperatorId(context.Background(), ec.operatorAddr)
 	if err != nil {
-		ec.logger.Error("Failed to get operator stake", "err", err)
+		// we only log and don't return because we might eventually add more metrics below,
+		// and returning early would introduce a bug that we would be skipping those other metrics
+		ec.logger.Error("Failed to get operator id", "err", err)
+	} else if operatorId == [32]byte{} {
+		ec.logger.Warn("Operator is not registered. Skipping registeredStake metric collection.", "fn", "economicCollector.Collect")
 	} else {
-		for quorumIdx, quorumNum := range quorumNums {
-			// TODO: this is stupid.. when AVSs scale to have 5K operators we'll be running through a bunch of operators
-			// we should instead just call registryCoordinator.getQuorumBitmapIndicesByOperatorIdsAtBlockNumber
-			// and stakeRegistry.getStakeForOperatorIdForQuorumAtBlockNumber directly
-			for _, operator := range blsOperatorStateRetrieverOperator[quorumIdx] {
-				if operator.OperatorId == ec.operatorId {
-					stakeFloat64, _ := operator.Stake.Float64()
-					ch <- prometheus.MustNewConstMetric(
-						ec.registeredStake, prometheus.GaugeValue, stakeFloat64, strconv.Itoa(int(quorumNum)), ec.quorumNames[quorumNum],
-					)
+		quorumNums, blsOperatorStateRetrieverOperator, err := ec.avsRegistryReader.GetOperatorsStakeInQuorumsOfOperatorAtCurrentBlock(context.Background(), operatorId)
+		if err != nil {
+			ec.logger.Error("Failed to get operator stake", "err", err)
+		} else {
+			for quorumIdx, quorumNum := range quorumNums {
+				// TODO: this is stupid.. when AVSs scale to have 5K operators we'll be running through a bunch of operators
+				// we should instead just call registryCoordinator.getQuorumBitmapIndicesByOperatorIdsAtBlockNumber
+				// and stakeRegistry.getStakeForOperatorIdForQuorumAtBlockNumber directly
```

Comment:
Why don't we do these two per-operator fetches?

Reply:
Because I was lazy at the time. Fixed: [858ae16](https://github.com/Layr-Labs/eigensdk-go/pull/54/commits/858ae16cda0dcdcf98ec391c40af565a00c39edd)